centreon/src/Core/Dashboard/Application/UseCase/FindSingleMetaMetric/FindSingleMetaMetricResponse.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * Copyright 2005 - 2025 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Dashboard\Application\UseCase\FindSingleMetaMetric;
+
+final readonly class FindSingleMetaMetricResponse
+{
+    public function __construct(
+        public int $id,
+        public string $name,
+        public ?string $unit,
+        public ?float $currentValue,
+        public ?float $warningHighThreshold,
+        public ?float $warningLowThreshold,
+        public ?float $criticalHighThreshold,
+        public ?float $criticalLowThreshold
```

Comment:
if can be null, so you need to initalize the value
```suggestion
        public ?string $unit = null,
        public ?float $currentValue = null,
        public ?float $warningHighThreshold = null,
        public ?float $warningLowThreshold = null,
        public ?float $criticalHighThreshold = null,
        public ?float $criticalLowThreshold = null
```